src/components/Typography/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC, useEffect, useState } from 'react'
+
+import { IProps } from './core/typography.interfaces'
+import s from './core/typography.module.scss'
+
+export const Typography: FC<IProps> = ({
+  children,
+  className,
+  font,
+  size,
+  element = 'p',
+  weight,
+  color,
+}) => {
+  const [classNames, setClassNames] = useState<string[]>([])
+
+  useEffect(() => {
+    const classNames = []
+
+    if (font) {
+      classNames.push(s[`typography--font-${font}`])
+    }
+    if (weight) {
+      classNames.push(s[`typography--weight-${weight}`])
+    }
+    if (size) {
+      classNames.push(s[`typography--size-${size}`])
+    }
+    if (color) {
+      classNames.push(s[`typography--color-${color}`])
+    }
+    if (className) {
+      classNames.push(className)
+    }
+
+    setClassNames(classNames)
+  }, [element, font, size, weight, color, className])
+
+  switch (element) {
+    case 'p':
+      return <p className={classNames.join(' ')}>{children}</p>
+    case 'pre':
+      return <pre className={classNames.join(' ')}>{children}</pre>
```

Comment:
Just a suggestion to reduce duplication, feel free to ignore:

```suggestion
  const props = {
    className: classNames.join(' '),
    children,
  }
  
  switch (element) {
    case 'p':
      return <p {...props} />
    case 'pre':
      return <pre {...props} />
```